SignalMessaging/contacts/OWSSyncManager.m

```diff
@@ -307,6 +324,49 @@ - (AnyPromise *)syncContactsForSignalAccounts:(NSArray<SignalAccount *> *)signal
     return promise;
 }
 
+- (AnyPromise *)syncAllGroups
+{
+    NSMutableArray<TSGroupThread *> *groupThreads = @[].mutableCopy;
+    NSMutableArray<AnyPromise *> *promises = @[].mutableCopy;
+    [TSGroupThread enumerateCollectionObjectsUsingBlock:^(id obj, BOOL *stop) {
+        if (![obj isKindOfClass:[TSGroupThread class]]) {
+            if (![obj isKindOfClass:[TSContactThread class]]) {
```

Comment:
This check seems like it would be redundant?

Reply:
This was copied from the original code of Signal. I tried not to check this and make the `id obj` as `TSGroupThread *obj`, but I got crashes here. So I just left this code as it was.